src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java

```diff
@@ -81,6 +81,15 @@ public Oauth2Policy (OAuth2PolicyConfiguration oAuth2PolicyConfiguration) {
     public void onRequest(Request request, Response response, ExecutionContext executionContext, PolicyChain policyChain) {
         logger.debug("Read access_token from request {}", request.id());
 
+        final String resourceNameTemplate = oAuth2PolicyConfiguration.getOauthResource();
+
+        if(resourceNameTemplate != null){
```

Comment:
I do not see any reason for which the `resourceNameTemplate` would be null. Can you explain a bit please?

Reply:
It's only for passing tests. It can be null in some tests.

Reply:
As per https://github.com/gravitee-io/gravitee-policy-oauth2/blob/master/src/main/resources/schemas/schema-form.json#L45 resource is required so I assume tests are wrong and resource is always provided.

Reply:
Yes, right, I see now that I can fix tests and reduce code in policy. I will update PR a little bit later.